projects/vdk-core/src/vdk/internal/builtin_plugins/run/datajob_initializer.py

```diff
@@ -41,15 +41,15 @@ def initialize_job(self, context: JobContext):
         context.step_builder.add_step(step)
 
 
-class InitialisedJobInput:
+class InitialisedJobContext:
     """
-    A contextmanager that executes all the necessary "setup" logic needed to return an initialised IJobInput
+    A contextmanager that executes all the necessary "setup" logic needed to return an initialised JobContext
+    and finalize logic when the context is exited
 
     Sample usage::
 
-        datajob_directory = Path(abspath(file_relative_path(__file__, '../')))
-        with InitialisedJobInput(datajob_directory) as vdk_job_input:
-            #... use vdk_job_input object to interact with SuperCollider
+        with InitialisedJobContext(datajob_directory) as vdk_job_context:
+            #... use vdk_job_context.job_input object to interact with SuperCollider
```

Comment:
What do you think about returning the `JobContext` object rather than just the `JobInput` ?

Related, what do you think about the name `InitialisedJobContext`?

Reply:
Are there any use-cases that require JobContext? It's too broad interface. I think we should expose the minimum necessary interface surface otherwise we are in hard time to maintain it. 

Also it's not very mature.  Unlike JobInput it was created fairly recently. In the future, it's possible to make it Immutable (once initialize_job phase is finished) to prevent changes that might lead to unpredictable behaviour. In that case it might be fine.

Reply:
I like `StandaloneDataJob`.  I'll switch to that

Reply:
> Are there any use-cases that require JobContext?

I found one (although perhaps this is possible via the IJobInput interface?)

After calling `send_tabular_data_for_ingestion` I want to wait for the ingestor to finish sending the messages.

The only "cleanish" way I've found to do this is via the ingester attribute on JobContext.

```
  self.vdk_job_context.job_input.send_tabular_data_for_ingestion(
            rows=...snip...
        )
        # Ensure the ingestor send queue is emptied before we exit
        self.log.info(f"Waiting for VDK ingestor to finish sending queued messages")
        cast(IngesterRouter, self.vdk_job_context.ingester).close()
```       

Now that I think of it however, I could perhaps put a call to `ingester.close()` during the `__exit__` method, ala:

```
def __exit__(self):
        cast(IngesterRouter, self.__job.job_context.ingester).close()
        self.__job.__exit()
```       

WDYT?

Reply:
This is possible by using [`wait_to_finish_after_every_send`](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_configuration.py#L108) configuration. 

Here is an example job where it is used : https://github.com/vmware/versatile-data-kit/blob/main/examples/online-exhibition/online-exhibition/config.ini#L33

Reply:
Aha!  Thats a much better way to do it.

I'll switch that on and revert to just exposing the `JobInput`

Reply:
I've addressed all the points in this conversation.  Resolving